pkg/monitortests/node/kubeletselinuxlabels/monitortest.go

```diff
@@ -0,0 +1,166 @@
+				HumanMessage(msgPhaseOSUpdate).
+				WithAnnotation(monitorapi.AnnotationConstructed, monitorapi.ConstructionOwnerNodeLifecycle).
+				WithAnnotation(monitorapi.AnnotationRoles, roles).
+				WithAnnotation(monitorapi.AnnotationPhase, "OperatingSystemUpdate")
+			intervals = append(intervals, nodeStateTracker.CloseIfOpenedInterval(nodeLocator, osUpdateState,
+				statetracker.SimpleInterval(monitorapi.SourceNodeState, monitorapi.Info, osUpdateMB),
+				event.From)...)
+			nodeStateTracker.OpenInterval(nodeLocator, rebootState, event.From)
+		case "Starting":
+			// Not ported, so we don't have a Source to check
+			mb := monitorapi.NewMessage().Reason(monitorapi.NodeUpdateReason).
+				HumanMessage(msgPhaseDrain).
+				WithAnnotation(monitorapi.AnnotationConstructed, monitorapi.ConstructionOwnerNodeLifecycle).
+				WithAnnotation(monitorapi.AnnotationRoles, roles).
+				WithAnnotation(monitorapi.AnnotationPhase, "Drain")
+			intervals = append(intervals, nodeStateTracker.CloseIfOpenedInterval(nodeLocator, drainState,
+				statetracker.SimpleInterval(monitorapi.SourceNodeState, monitorapi.Info, mb),
+				event.From)...)
+
+			osUpdateMB := monitorapi.NewMessage().Reason(monitorapi.NodeUpdateReason).
+				HumanMessage(msgPhaseOSUpdate).
+				WithAnnotation(monitorapi.AnnotationConstructed, monitorapi.ConstructionOwnerNodeLifecycle).
+				WithAnnotation(monitorapi.AnnotationRoles, roles).
+				WithAnnotation(monitorapi.AnnotationPhase, "OperatingSystemUpdate")
+			intervals = append(intervals, nodeStateTracker.CloseIfOpenedInterval(nodeLocator, osUpdateState,
+				statetracker.SimpleInterval(monitorapi.SourceNodeState, monitorapi.Info, osUpdateMB),
+				event.From)...)
+
+			rebootMB := monitorapi.NewMessage().Reason(monitorapi.NodeUpdateReason).
+				HumanMessage(msgPhaseReboot).
+				WithAnnotation(monitorapi.AnnotationConstructed, monitorapi.ConstructionOwnerNodeLifecycle).
+				WithAnnotation(monitorapi.AnnotationRoles, roles).
+				WithAnnotation(monitorapi.AnnotationPhase, "Reboot")
+			intervals = append(intervals, nodeStateTracker.CloseIfOpenedInterval(nodeLocator, rebootState,
+				statetracker.SimpleInterval(monitorapi.SourceNodeState, monitorapi.Info, rebootMB),
+				event.From)...)
+		}
+	}
+	// Close all node intervals left hanging open:
+	intervals = append(intervals, nodeStateTracker.CloseAllIntervals(locatorToMessageAnnotations, end)...)
+
+	return intervals, nil
+}
+
+func (lw *labelWatcher) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	testCaseJunit := junitapi.JUnitTestCase{Name: testName}
+	kubeClient, err := kubernetes.NewForConfig(lw.adminRESTConfig)
+	if err != nil {
+		return nil, err
+	}
+	nodes, err := kubeClient.CoreV1().Nodes().List(ctx, v1.ListOptions{})
+	if err != nil {
+		return []*junitapi.JUnitTestCase{&testCaseJunit}, err
+	}
+	oc := exutil.NewCLIWithoutNamespace("default").AsAdmin()
+	for _, val := range nodes.Items {
```

Comment:
Let's make this parallel with a sync.WaitGroup.

Reply:
Is that a premature optimization though? In the CI are we testing with lots of nodes?

Reply:
Perhaps. We usually test with only 3 workers and 3 masters. Not that big of a deal. We can skip the waitgroup.

Reply:
I'll see if I can get it working. I'm going to focus on making sure this works before that though.